src/authentication/components/AuthConnector.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react'
+import { auth } from 'src/util/remote-functions-background'
+import { getRemoteEventEmitter } from 'src/util/webextensionRPC'
+import {
+    AuthenticatedUser,
+    AuthenticatedUserWithClaims,
+} from 'src/authentication/background/types'
+import { UserFeatures } from 'firebase-backend/firebase/functions/src/types'
+
+interface State {
+    currentUser: AuthenticatedUserWithClaims | AuthenticatedUser | null
+    authorizedFeatures: UserFeatures[]
+}
+
+// tslint:disable-next-line:variable-name
+export function withCurrentUser(WrappedComponent) {
```

Comment:
If I see correctly, this break type-safety, right? Would be great if we could keep that, makes refactoring a lot easier.